components/resourceDetails/resourceSharing/addAgentButton/agentPickerModal/index.jsx

```diff
@@ -109,31 +119,43 @@ export const handleConfirmation = ({
       setConfirmed(null);
       setOpen(null);
       setConfirmationSetup(false);
+      setContent(null);
+      setTitle(null);
     }
   };
 };
 
-export const handleSaveContact = async (iri, people, addressBook, fetch) => {
+export const handleSaveContact = async (iri, contacts, addressBook, fetch) => {
+  const {
+    dataset: addressBookDataset,
+    containerUrl: addressBookContainerUrl,
+  } = addressBook;
   let error;
   if (!iri) {
     return;
   }
 
   try {
     const { name, webId, types } = await fetchProfile(iri, fetch);
-    const existingContact = await findContactInAddressBook(
-      people,
+    // TODO: we will likely need to change this if we want to save groups on the fly as well
+    const existingContact = await findPersonContactInAddressBook(
+      addressBook,
       webId,
       fetch
     );
-
     if (existingContact.length) {
       return;
     }
 
     if (name) {
       const contact = { webId, fn: name };
-      await saveContact(addressBook, contact, types, fetch);
+      await saveContact(
```

Comment:
Yeah, I decided to keep using this one as long as it works until we switch all the code in Contacts instead of modifying this function and breaking the functionality in the Contacts page. I think we should have a separate ticket to do all this work.